src/core/Bot.py

```diff
@@ -261,19 +264,32 @@ async def db_latency(self):
         return f"{t2*1000:.2f} ms"
 
     @staticmethod
-    async def getch(get_method, fetch_method, _id):  # why does c have all the fun?
+    async def getch(get_method, fetch_method, _id) -> Any:  # why does c have all the fun?
         try:
             _result = get_method(_id) or await fetch_method(_id)
         except (discord.HTTPException, discord.NotFound):
             return None
         else:
             return _result
 
+    async def get_or_fetch_message(self, channel: discord.Channel, message_id: int) -> Optional[discord.Message]:
+        # caching cause, due to rate limiting 50/1
+        try:
+            return self.message_cache[message_id]
+            # scripting is always fater than `.get()`
+        except KeyError:
+            before = discord.Object(message_id + 1)
+            after = discord.Object(message_id - 1)
+            async for msg in channel.history(limit = 1, before=before, after=after):
+                if msg:
+                    self.message_cache[msg.id] = msg
+                    return msg
+
     async def send_message(self, channel_id, content, **kwargs):
         await self.http.send_message(channel_id, content, **kwargs)
 
     async def convey_important_message(
-        self, guild: discord.Guild, text: str, *, view=None, title="⚠️__**IMPORTANT**__⚠️"
+        self, guild: discord.Guild, text: str, *, view=None, title="\N{WARNING SIGN}__**IMPORTANT**__\N{WARNING SIGN}"
```

Comment:
I changed cause my IDE wasn't able to parse that emoji :weary:

Reply:
Mine too 😂